avocado/utils/network/interfaces.py

```diff
@@ -128,6 +128,74 @@ def add_ipaddr(self, ipaddr, netmask):
         except Exception as ex:
             raise NWException("Failed to add address {}".format(ex))
 
+    @property
+    def vlans(self):
+        """Return a list of vlans"""
+        vlans = {}
+        with open('/proc/net/vlan/config', encoding="utf-8") as vlan_config_file:
+            for line in vlan_config_file:
+                # entry is formatted as "vlan_name | vlan_id | parent_device"
+                line = line.strip().replace(" ", "")
```

Comment:
You don't need both `strip()` and `replace(" ", "")`, just the `replace(" ", "")` should do it.

Reply:
`.strip()` removes the newline character of the current line and it won't match with `.endswith(self.name)` since `self.name` doesn't have the newline char. Another approach would be `line = "".join(line.split())`...

Reply:
> `.strip()` removes the newline character of the current line and it won't match with `.endswith(self.name)` since `self.name` doesn't have the newline char. Another approach would be `line = "".join(line.split())`...

You're right, please disregard my comment.